File: BoxQuizeApp/app/src/main/java/ru/sport/boxquizeapp/QuizViewModel.java
package ru.sport.boxquizeapp;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

public class QuizViewModel extends ViewModel {
    private QuizRepository quizRepository;
    private LiveData<List<Question>> questionsLiveData;

    public QuizViewModel() {
        quizRepository = new QuizRepository();
        questionsLiveData = quizRepository.getQuestions();
    }

    public LiveData<List<Question>> getQuestionsLiveData() {
        return questionsLiveData;
    }
}

File: KTprila2/app/src/main/java/com/sport/ktprila/QuizViewModel.kt
package com.sport.ktprila

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class QuizViewModel : ViewModel() {
    private val quizRepository = QuizRepository()
    val quizQuestions = MutableLiveData<List<Question>>()

    fun loadQuizQuestions() {
        CoroutineScope(Dispatchers.IO).launch {
            val questions = quizRepository.getQuizQuestions()
            quizQuestions.postValue(questions)
        }
    }
}
